### PR TITLE
add nativefiledialog-extended

### PR DESCRIPTION
**Pull request by xq114**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/c/catch2/xmake.lua [hunk 1]
```diff
@@ -40,6 +40,9 @@ package("catch2")
 
     on_component("main", function (package, component)
         component:add("links", "Catch2Main")
+        if package:is_plat("windows") then
+            component:add("ldflags", "-subsystem:console")
```
Comment (SirLynix) [inline packages/c/catch2/xmake.lua:44]: why would catch2 force a console? people may want to use it in a GUI app and redirect stdout to widgets or a file

Comment (xq114) [reply packages/c/catch2/xmake.lua:44]: The "main" component contains a `main` function, which is expected to be linked into a console app. For those who want a custom entry point they should only link to the `lib` component. See https://github.com/catchorg/Catch2/blob/devel/docs/cmake-integration.md#cmake-targets

**Comment by xq114** [comment]
freebsd上安装失败了，不影响

File: packages/n/nativefiledialog-extended/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    if is_plat("windows") then
+        add_syslinks("shell32", "ole32")
+    elseif is_plat("macosx") then
+        add_frameworks("AppKit", "UniformTypeIdentifiers")
+    end
+    on_load("linux", function (package)
+        if package:config("portal") then
+            package:add("deps", "dbus")
+        else
+            package:add("deps", "gtk-3")
```
Comment (waruqi) [inline packages/n/nativefiledialog-extended/xmake.lua:25]: maybe `gtk+3`

**Pull request merged**